### ISSUE TITLE
Hide Player is detected by target (TAB)

### ISSUE THREAD

**Issue by helenatodev** [AscensionGameDev/Intersect-Engine#781]
**Describe the bug**

**Version Encountered**
Build: 0.7.0.121

**To Reproduce**
1 - Create an event that makes the player hidden, hidden. with the hide player function.
Other players can detect you by selecting TAB
3 - log in with another character at the same time
2- See the error

**Expected behavior**
I believe this is a good tool for the game team, being able to hide and inspect the player anonymously, but it becomes totally useless as it can be detected with the TAB key.

**Operating System:**
Windows, Mac, and Linux

EDIT-

The player must be invulnerable by becoming invisible, because he would become an invisible entity.
This bug works on maps where pvp between players is allowed.
In short, the function of staying invisible, should block identification by TAB and also not take damage with area spells, the player can find the GM using area spells.

**Comment by Xenepherth** [comment]
My TAB don't work that but i can't tab selection when a player be stealth

**Comment by helenatodev** [comment]
You need to leave your Arena or Normal map, it cannot be Safe

**Comment by WeylonSantana** [comment]
@Xenepherth from what I read, the @helenatodev it's talking about the function of events to hide the player and not the spell of making the player stealth, i have tested this bug and it actually happens, if the map is not safe (where you can select other players with tab), pressing tab will select a hidden player, which has been hidden by events. On a safe map, you can still select the hidden player, just be lucky enough to hover the mouse over the hidden player

**Comment by Xenepherth** [comment]
Idk my client don't do that sometimes i'm fixing code bugs by myself...

**Comment by Xenepherth** [comment]
yea i know why that

Solution: 
Client/Entities/Player.cs

Go to DrawTargets

and

`foreach (var en in Globals.Entities)
            {
                if (en.Value == null)
                {
                    continue;
                }

                if (!en.Value.IsStealthed() || !(en.Value is Player player && Globals.Me.IsInMyParty(player)))
                {
                    if (en.Value.GetType() != typeof(Projectile) && en.Value.GetType() != typeof(Resource))
                    {
                        if (TargetType == 0 && TargetIndex == en.Value.Id)
                        {       
                                en.Value.DrawTarget((int)TargetTypes.Selected);
                        }
                    }
                }
                else
                {
                    //TODO: Completely wipe the stealthed player from memory and have server re-send once stealth ends.
                    ClearTarget();
                }
            }

            foreach (MapInstance eventMap in MapInstance.Lookup.Values)
            {
                foreach (var en in eventMap.LocalEntities)
                {
                    if (en.Value == null)
                    {
                        continue;
                    }

                    if (en.Value.CurrentMap == eventMap.Id &&
                        !((Event) en.Value).DisablePreview &&
                        (!en.Value.IsStealthed() || !(en.Value is Player player && Globals.Me.IsInMyParty(player))))
                    {
                        if (TargetType == 1 && TargetIndex == en.Value.Id)
                        {
                            en.Value.DrawTarget((int) TargetTypes.Selected);
                        }
                    }
                }
            }
`

**Comment by blinkuz** [comment]
@Xenepherth Perhaps a PR on your part is better sharing the solution, it is a simple line of code that you add but can help solve the problem for everyone.

**Comment by Xenepherth** [comment]
> Perhaps a PR on your part is better sharing the solution, it is a simple line of code that you add but can help solve the problem for everyone.

my local change is so much if you can a PR do that bro...

**Comment by Cheshire92** [comment]
And that's why you probably shouldn't build your game on a direct fork. 🤣

**Comment by Xenepherth** [comment]
> And that's why you probably shouldn't build your game on a direct fork. 🤣

idk xD

**Comment by Arufonsu** [comment]
> 
> 
> yea i know why that
> 
> Solution:
> Client/Entities/Player.cs

That could easily be reversed if the game client gets modified/"hacked". Wouldn't we need a server side solution too (?)

As for PRs and personal changes, you may want to keep a change log of ur stuff, and if you happen to fix anything that can be generally used in the engine, PRs for them are always a nice way to contribute/discuss <3

Regards.
